src/core/rewrite/execute.rs

```diff
@@ -954,9 +971,14 @@ pub struct ExecuteRebasePlanOptions<'a> {
 
 /// The result of executing a rebase plan.
 #[must_use]
+#[derive(Debug)]
 pub enum ExecuteRebasePlanResult {
     /// The rebase operation succeeded.
-    Succeeded,
+    Succeeded {
+        /// Mapping from old OID to new/rewritten OID. Will always be empty for on disk rebases.
+        // TODO should this be an Option?
+        rewritten_oids: HashMap<NonZeroOid, MaybeZeroOid>,
+    },
```

Comment:
Regarding this `TODO`: only in mem rebases will be using this property, b/c on disk rebases don't have access to (as far as I can tell) the list of rewritten commits, and are currently only ever using `Default::default()` for this property, but that could leave us unable to tell if a `Succeeded` result actually rewrote 0 commits, or if it's just a misleading results from an on disk rebase. Using an `Option<...>` here could solve that b/c we can then say that all on-disk rebases use `None`.

Ok, so that said, I like the `Option<...>` feels more correct, but given the very limited usage of this (at this time), I suspect that this is OK as is. What do you think?

Reply:
> on disk rebases don't have access to (as far as I can tell) the list of rewritten commits

The on-disk rebases use the `post-rewrite` hook to get access to this information, but it's a complicated workflow, and the information isn't returned synchronously. I think using an `Option` here is warranted to make that clear, and we can `unwrap` it on the other end.

Reply:
This is an `Option` now.